src/figuresClasses.ts

```diff
@@ -1,11 +1,96 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape;
```

Comment:
You can just have 

```suggestion
  shape = 'triangle';
```
So you don't need to set it in the constructor
This can also be fixed in the other classes